lib/web/apiserver.go

```diff
@@ -349,12 +351,15 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.POST("/webapi/u2f/password/changerequest", h.WithAuth(h.u2fChangePasswordRequest))
 	h.POST("/webapi/u2f/signrequest", httplib.MakeHandler(h.mfaChallengeRequest))
 	h.POST("/webapi/u2f/signrequest/token", httplib.MakeHandler(h.getMFAChallengeRequestWithTokenHandle))
+	h.POST("/webapi/u2f/signrequest/auth", h.WithAuth(h.getMFAChallengeRequestWithAuthHandle))
```

Comment:
Had trouble naming this, any suggestions welcome. There are 3 types of `u2f/signrequest`:

`h.POST("/webapi/u2f/signrequest", httplib.MakeHandler(h.mfaChallengeRequest))`: public handler where auth is their username and password during login

`h.POST("/webapi/u2f/signrequest/token", httplib.MakeHandler(h.getMFAChallengeRequestWithTokenHandle))`: public handler where auth is the token, used during account recovery

`h.POST("/webapi/u2f/signrequest/auth", h.WithAuth(h.getMFAChallengeRequestWithAuthHandle))`: private handler where auth is that the requestor == currently logged in user

Reply:
I'll review properly later, but in relation to this we've got an ongoing RFD where I'm aliasing (and eventually moving) /u2f/ endpoints to more generic /mfa/ counterparts, which is applicable here too.

Please have a look at the Login and Registration API sections and let me know what you think: [WebAuthn RFD](https://github.com/gravitational/teleport/pull/7808).